core/vm/contract.go

```diff
@@ -86,11 +88,11 @@ func (c *Contract) validJumpdest(dest *uint256.Int) bool {
 	udest, overflow := dest.Uint64WithOverflow()
 	// PC cannot go beyond len(code) and certainly can't be bigger than 63bits.
 	// Don't bother checking for JUMPDEST in that case.
-	if overflow || udest >= uint64(len(c.Code)) {
+	if overflow || udest >= c.CodeSize() {
 		return false
 	}
 	// Only JUMPDESTs allowed for destinations
-	if OpCode(c.Code[udest]) != JUMPDEST {
+	if OpCode(c.Code[c.CodeBeginOffset()+udest]) != JUMPDEST {
```

Comment:
Maybe something like this, and internalize the call to CodeBeginOffset?
```suggestion
	if c.OpCodeAt(udest) != JUMPDEST {
```

Reply:
I went with @axic's suggestion below and used existing `GetOp` for this.